home/models.py

```diff
@@ -24,18 +25,51 @@
 )
 
 
+@register_setting
+class SiteSettings(BaseSetting):
+    content_variations_options = StreamField(
+        [
+            ("content_variations_options", blocks.CharBlock()),
+        ],
+        blank=True,
+        null=True,
+    )
+
+
 class MediaBlock(AbstractMediaChooserBlock):
     def render_basic(self, value, context=None):
         pass
 
 
+def get_choices():
+    if SiteSettings.objects.first():
+        return [
+            (choice, choice)
+            for choice in SiteSettings.objects.first().content_variations_options
+        ]
+    return []
+
+
+class VariationBlock(blocks.StructBlock):
+    variation_message = blocks.ChoiceBlock(
+        required=False,
+        choices=get_choices,
+        default=None,
+        help_text="Add variation fields to the site settings if you'd like to see them here",
+    )
+    message = blocks.TextBlock(
+        max_lenth=4096, help_text="each message cannot exceed 4096 characters."
+    )
+
+
 class WhatsappBlock(blocks.StructBlock):
     image = ImageChooserBlock(required=False)
     document = DocumentChooserBlock(icon="document", required=False)
     media = MediaBlock(icon="media", required=False)
     message = blocks.TextBlock(
         max_lenth=4096, help_text="each message cannot exceed 4096 characters."
```

Comment:
something to note for future is with the new changes, we should add check so that if message has media, the char limit is lower